surefire-providers/surefire-junit-platform/src/test/java/org/apache/maven/surefire/junitplatform/JUnitPlatformProviderTest.java

```diff
@@ -891,6 +893,57 @@ public void parsesConfigurationParameters()
         assertEquals( "EOF", provider.getConfigurationParameters().get( "qux" ) );
     }
 
+    @Test
+    public void shouldFilterTestMethod()
+    {
+        ProviderParameters providerParameters = providerParametersMock();
+        TestListResolver testListResolver = new TestListResolver( "**/*Test#test*" );
+        assertFalse( testListResolver.isEmpty() );
+        assertFalse( testListResolver.isWildcard() );
+        TestRequest request = new TestRequest( null, null, testListResolver, 0 );
+        when( providerParameters.getTestRequest() ).thenReturn( request );
+
+        JUnitPlatformProvider provider = new JUnitPlatformProvider( providerParameters );
+
+        assertEquals( provider.getFilters().length, 1 );
+
+        assertThat( provider.getFilters()[0] )
+            .isInstanceOf( TestMethodFilter.class );
+
+        assertThat( (TestListResolver) getInternalState( provider.getFilters()[0], "testListResolver" ) )
```

Comment:
is cast needed?

Reply:
yes, it is, IDEA would mark the line red.
See the generics in the retunred type in `getInternalState()`.

Reply:
I can add one more assertThat.isInstanceOf.

Reply:
done